src/bootstrap/src/core/config/config.rs

```diff
@@ -1511,8 +1513,19 @@ impl Config {
             metrics: _,
             android_ndk,
             optimized_compiler_builtins,
+            jobs,
         } = toml.build.unwrap_or_default();
 
+        if config.jobs.is_none() {
```

Comment:
Or, I could `match (flags.jobs, toml_job)` here too if that's easier to follow.

Reply:
What about this:

```diff
diff --git a/src/bootstrap/src/core/config/config.rs b/src/bootstrap/src/core/config/config.rs
index abfc6178ce7..2f1334a9f38 100644
--- a/src/bootstrap/src/core/config/config.rs
+++ b/src/bootstrap/src/core/config/config.rs
@@ -1290,7 +1290,6 @@ pub(crate) fn parse_inner(
         config.rustc_error_format = flags.rustc_error_format;
         config.json_output = flags.json_output;
         config.on_fail = flags.on_fail;
-        config.jobs = flags.jobs.filter(|j| *j > 0);

         config.cmd = flags.cmd;
         config.incremental = flags.incremental;
@@ -1516,15 +1515,7 @@ fn get_table(option: &str) -> Result<TomlConfig, toml::de::Error> {
             jobs,
         } = toml.build.unwrap_or_default();

-        if config.jobs.is_none() {
-            config.jobs = match jobs {
-                Some(jobs) if jobs > 0 => Some(jobs),
-                _ => Some(
-                    std::thread::available_parallelism().map_or(1, std::num::NonZeroUsize::get)
-                        as u32,
-                ),
-            };
-        }
+        config.jobs = Some(threads_from_config(flags.jobs.unwrap_or(jobs.unwrap_or(0))));

         if let Some(file_build) = build {
```

Reply:
Works for me. I'll change it tmrw.

Reply:
Modified to use this, gonna wait for CI to test it because the bootstrap tests hangs for me locally on msvc lol